### PR TITLE
[r=Rocky] Carl composite lit check

### PR DESCRIPTION
**Pull request by 0xfaded**
Type checks for arrays, slices and structs

### CODE REVIEW

File: checkexpr.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func CheckExpr(ctx *Ctx, expr ast.Expr, env *Env) (Expr, []error) {
 	case *ast.KeyValueExpr:
 		return checkKeyValueExpr(ctx, expr, env)
 	default:
+		panic("grr")
```
Comment (0xfaded) [inline checkexpr.go]: Remove this

File: checkcompositelit.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"go/ast"
 )
 
+/*
 func checkCompositeLit(ctx *Ctx, lit *ast.CompositeLit, env *Env) (aexpr *CompositeLit, errs []error) {
```
Comment (0xfaded) [inline checkcompositelit.go]: Remove commented old code

**Comment by 0xfaded** [comment]
Hey Rocky,

Don't merge this one yet until I get around to addressing the above comments.

**Comment by 0xfaded** [comment]
comments addressed

**Comment by rocky** [comment]
Ok. 

The regression that bothers me the most from a usuability standpoint is where we can't valuate package variables.

**Comment by 0xfaded** [comment]
Merged

**Pull request closed without merging**